lookout/style/format/analyzer.py

```diff
@@ -163,25 +165,73 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
                                 for fix_index, line_fix in enumerate(file_fix.line_fixes))
         return comments
 
+    @classmethod
+    def check_training_required(
+            cls, old_model: FormatModel, ptr: ReferencePointer, config: Mapping[str, Any],
+            data_service: "lookout.core.data_requests.DataService", **data) -> bool:
+        """
+        Return True if the format model needs to be refreshed; otherwise, False.
+
+        We calculate the ratio of the number of changed lines to the overall number of lines.
+        If it is bigger than lines_ratio_train_trigger - we need to train.
+
+        :param old_model: Current FormatModel.
+        :param ptr: Git repository state pointer.
+        :param config: configuration dict.
+        :param data: contains "files" - the list of files in the pointed state.
+        :param data_service: connection to the Lookout data retrieval service.
+        :return: True or False
+        """
+        _log = logging.getLogger(cls.__name__)
+        changes = list(request_changes(
+            data_service.get_data(), old_model.ptr, ptr, contents=True, uast=False))
+        base_files_by_lang = files_by_language(c.base for c in changes)
+        head_files_by_lang = files_by_language(c.head for c in changes)
+        config = cls._load_train_config(config)
+        for language, head_files in head_files_by_lang.items():
+            try:
+                lang_config = config[language]
+            except KeyError:
+                _log.warning("language %s is not supported, skipped", language)
+                continue
+            overall_lines = changed_lines = 0
+            for file in filter_files(head_files, lang_config["line_length_limit"],
+                                     lang_config["overall_size_limit"], log=_log):
+                head_lines = len(file.content.splitlines())
+                overall_lines += head_lines
+                try:
+                    prev_file = base_files_by_lang[language][file.path]
+                except KeyError:
+                    changed_lines += head_lines
+                else:
+                    changed_lines += len(find_new_lines(prev_file, file))
```

Comment:
we should probably check for deletions also (with `find_deleted_lines`) since a 20% change in this direction (less data) would probably also require a retrain.